script.js

```diff
@@ -1,4 +1,348 @@
+    while (drawCounter < 2) {
+      playerCards.push(playerDeck.pop());
+      computerCards.push(computerDeck.pop());
+      drawCounter = drawCounter + 1;
+    }
+
+    // calculate total score for each player
+    playerScore = calculatePlayerScore();
+    computerScore = calculateComputerScore();
+    console.log("playerScore: " + playerScore);
+    console.log("computerScore: " + computerScore);
+
+    myOutputValue =
+      "You drew a " +
+      playerCards[0].name +
+      " and " +
+      playerCards[1].name +
+      ". Your score = " +
+      playerScore +
+      "<br><br>Computer drew " +
+      computerCards[0].name +
+      " and " +
+      computerCards[1].name +
+      ". Computer score = " +
+      computerScore +
+      "<br><br> Enter 'hit' to draw another card, or 'stand' to end your turn.";
+
+    currentGameMode = "hit or stand";
+
+    // If computer score < 17, computer has to draw card again
+    if (computerScore < 17) {
+      myOutputValue =
+        "You drew a " +
+        playerCards[0].name +
+        " and " +
+        playerCards[1].name +
+        ". Your score = " +
+        playerScore +
+        "<br><br>Computer drew " +
+        computerCards[0].name +
+        " and " +
+        computerCards[1].name +
+        ". Computer score = " +
+        computerScore +
+        "<br><br> Computer has to draw again. Click submit to continue.";
+
+      currentGameMode = "computer draws";
+    }
+  } else if (currentGameMode == "hit or stand") {
+    // If player chooses to stand
+    if (input == "stand") {
+      computerScore = calculateComputerScore();
+      playerScore = calculatePlayerScore();
+      myOutputValue = endGame(playerScore, computerScore);
+
+      currentGameMode = "draw card";
+      playerCards = [];
+      computerCards = [];
+    }
+    // If player chooses to hit
+    if (input == "hit") {
+      playerCards.push(playerDeck.pop());
+      playerScore = calculatePlayerScore();
+      computerScore = calculateComputerScore();
+
+      var newPlayerIndex = playerCards.length - 1;
+
+      myOutputValue =
+        "You drew " +
+        playerCards[newPlayerIndex].name +
+        ". Your score = " +
+        playerScore +
+        ". Computer score = " +
+        computerScore +
+        "<br><br> Enter 'hit' to draw another card, or 'stand' to end your turn.";
+
+      console.log("player score: " + playerScore);
+      console.log("computer score" + computerScore);
+
+      // If player's new score exceeds 21, game automatically ends.
+      if (playerScore > 21) {
+        myOutputValue =
+          "You drew " +
+          playerCards[newPlayerIndex].name +
+          ". Your score = " +
+          playerScore +
+          ". Computer score = " +
+          computerScore +
+          "<br><br> Yikes! You lost, your score exceeded 21. Better luck next time! <br><br> Click submit to play again.";
+        currentGameMode = "draw card";
+        playerCards = [];
+        computerCards = [];
```

Comment:
This is the logic for resetting the game right? You can create a function called `resetGame` like so:

```
var resetGame = function () {
  currentGameMode = "draw card";
  playerCards = [];
  computerCards = [];
}
```

Then the following lines:
1. [205 to 207](https://github.com/rocketacademy/basics-blackjack/blob/a122772c1c05a29aae43234d1caabdbe74894d05/script.js#L205-L207)
2. [239 to 241](https://github.com/rocketacademy/basics-blackjack/blob/a122772c1c05a29aae43234d1caabdbe74894d05/script.js#L239-L241)
3. [283 to 285](https://github.com/rocketacademy/basics-blackjack/blob/a122772c1c05a29aae43234d1caabdbe74894d05/script.js#L283-L285)

... can be replaced with just 1 line each:

```
resetGame();
```

This will help to keep your program shorter by maybe a few lines, but it will prevent you from repeating these 3 lines over and over whenever you need to reset the game.